marbles/core/marbles/core/marbles.py

```diff
@@ -339,7 +339,10 @@ def _format_local(cls, name, value):
 
     @classmethod
     def _format_locals(cls, locals_):
-        return '\n'.join(cls._format_local(k, v) for k, v in locals_.items())
+        local_variable_names, local_variable_values = zip(*locals_.items())
```

Comment:
First, thanks for submitting your CLA 😄

A couple of notes:

**Empty Locals Dict**
If `locals_` is an empty dict, this variable assignment is going to fail:
```python
>>> empty_dict = dict()
>>> k, v = zip(*emtpy_dict.items())
----------------------------------------------------------------------
Traceback (most recent call last):
      1 empty_dict = dict()
----> 2 k, v = zip(*empty_dict.items())
ValueError: not enough values to unpack (expected 2, got 0)
```
**Variables and Variable Names**
These lines are a bit too long, which is causing the flake8 build step to fail. Could we use more concise variable names, and make sure to run the linter locally before pushing? You can see the docs on how to lint with flake8 and/or tox locally [here](https://marbles.readthedocs.io/en/stable/maintaining.html#linting).

**_common_shorten_repr**
I'm reluctant to rely on an internal function. What do you think about using [`unittest.util.safe_repr`](https://github.com/python/cpython/blob/3.7/Lib/unittest/util.py#L45) with `short=True` instead of `_common_shorten_repr`? If you already considered this and chose `_common_shorten_repr` anyway, I'd love to hear your thought process (and see it captured in an inline comment).

**Move Logic Into `_format_local`**
What do you think about moving this logic into `_format_local` classmethod instead? I think that would be simpler and is a more natural place to do this shortening. If we do the shortening there, we don't have to worry about unzipping and zipping the keys and values in `locals_` (because we're operating on one key, value pair at a time), and can avoid the issue where `locals_` is an empty dictionary entirely. We could also handle shortening multi-line locals and single-line locals separately (we already process them differently in `_format_local` anyway), which we might want to do. @leifwalsh, want to chime in?

Reply:
Awesome, thanks for the feedback!
I totally missed the part about linting, thanks for telling me. I will address that!

I definitely agree that it will be cleaner to put the logic in `_format_local`!

Do I suggest specific code changes in here or rather push a commit for it?